lighthouse-core/lib/url-shim.js

```diff
@@ -175,7 +175,7 @@ class URLShim extends URL {
 
 URLShim.URL = URL;
 
-URLShim.NON_NETWORK_PROTOCOLS = ['blob', 'data', 'intent'];
+URLShim.NON_NETWORK_PROTOCOLS = ['blob', 'data', 'intent', 'mailto', 'javascript'];
```

Comment:
I'm not sure this really applies to this list since we're checking `request.protocol` from the network records on this, maybe we should clarify our usage and we really should be maintaining another list?

i.e. if we want to check `new URL().protocol` we should be using the colon version, etc

Reply:
yeah, this one is kind of ambiguous (and the very similar `allowedProtocols` and `URLShim.isProtocolAllowed()` don't really clear things up).

Maybe it should be a method where we can at least have a method description/take a `NetworkRequest` so there's no confusing what it's checking?

Reply:
Since these items here are evaluated against the [devtools protocol's network](https://chromedevtools.github.io/devtools-protocol/tot/Network#type-Response) domain `protocol` string, lets rename it to `PROTOCOLS_NON_NETWORK_PROTOCOLS`. 


🤩🤩🤩